build/templates/_converters.py.mako

```diff
@@ -8,14 +8,70 @@ import datetime
 import six
 
 
+def _repeated_capability_string_to_list(channels, prefix):
+    '''Convert a string to a list of repeated capabilities numbers I.e. no prefix
+
+    This duplicates parsing in the driver, so if changes are made there, they will need to be replicated here.
+
+    '0,1,3' becomes [0, 1, 3]
+    '0-2' becomes [0, 1, 2]
+    'ScriptTrigger0,ScriptTrigger1' becomes [0,1]
+    '0,1,5-8,2' becomes [0, 1, 5, 6, 7, 8, 2]
+    '''
+    channels_list = []
+    for c in channels:
+        # We remove any prefix and change ':' to '-'
+        c = c.strip().replace(prefix, '').replace(':', '-')
+        cs = c.split('-')
+        if len(cs) > 1:
+            assert len(cs) == 2, "Only one '-' allowed. {0}".format(c)
+            start = cs[0]
+            end = cs[1]
+            if int(end) < int(start):
+                for i in range(int(start), int(end) - 1, -1):
+                    channels_list.append(str(i))
+            else:
+                for i in range(int(start), int(end) + 1):
+                    channels_list.append(str(i))
+        else:
+            channels_list.append(c)
+
+    return channels_list
+
+
 def convert_repeated_capabilities(repeated_capability, prefix=''):
+    '''Convert a repeated capabilities object to a comma delimited list
+
+    Args:
+        repeated_capability (str, list of str, list of int) -
+            str - single string that follows driver repeated capabilities format
+            list of str - each string must follow driver repeated capabilities format such that when joined with a comma is valid.
+                May or may not have prefix on each one. If not, the prefix will be added
+            list of int - list of str will be created by prefix + int
+            tuple - call back into this function with each item in tuple
+        prefix (str) - common prefix for all strings
+
+    Returns:
+        rep_cap (str) - string containing comma delimited list
+        rep_cal_list (list of str) - list of each repeated capability item with ranges expanded
+    '''
+    # If we recieved a tuple, then call ourselves with each item
+    if isinstance(repeated_capability, tuple):
+        rep_cap_list = []
+        rep_cap = []
+        for r in repeated_capability:
+            rc, rcl = convert_repeated_capabilities(r, prefix)
+            rep_cap.append(rc)
+            rep_cap_list += rcl
+        return ','.join(rep_cap), rep_cap_list
```

Comment:
I think you should return only one thing from this function, rep_cap_list.
`','.join()` is trivial to do on the client side. It adds little value to do it in the function and obscures the code.

Reply:
I have changed the logic for both `convert_repeated_capabilities()` and `_repeated_capability_string_to_list()`, including only returning the list